### PR TITLE
Support doctrine/annotations 2

### PR DESCRIPTION
**Pull request by derrabus**
WIP until doctrine/annotations 2 has been released.

This PR allows the installation of Annotations 2 and fixes all failing tests.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -78,6 +78,9 @@ jobs:
         run: "composer require doctrine/dbal ^${{ matrix.dbal-version }} --no-update"
         if: "${{ matrix.dbal-version != 'default' }}"
 
+      - name: "Uninstall PHPBench"
+        run: "composer remove --dev --no-update phpbench/phpbench"
```
Comment (greg0ire) [inline .github/workflows/continuous-integration.yml:82]: I'm looking into adding support for v2 to phpbench

Comment (greg0ire) [reply .github/workflows/continuous-integration.yml:82]: https://github.com/phpbench/phpbench/pull/1013

I ran the tests locally, and apart from an unrelated failure, all is well.

File: lib/Doctrine/ORM/Configuration.php [hunk 2]
```diff
@@ -182,17 +183,26 @@ public function newDefaultAnnotationDriver($paths = [], $useSimpleAnnotationRead
             );
         }
 
-        AnnotationRegistry::registerFile(__DIR__ . '/Mapping/Driver/DoctrineAnnotations.php');
+        if (method_exists(AnnotationRegistry::class, 'registerFile')) {
+            AnnotationRegistry::registerFile(__DIR__ . '/Mapping/Driver/DoctrineAnnotations.php');
+        }
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Configuration.php]: This line could be removed instead, couldn't it? https://github.com/doctrine/annotations/pull/271 ?

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (derrabus) [reply .github/workflows/continuous-integration.yml:82]: I think, we can live with this hack in our CI until PHPBench has merged and released your PR.

File: lib/Doctrine/ORM/Configuration.php (hunk 2 above)
Comment (derrabus) [reply lib/Doctrine/ORM/Configuration.php]: I guess we could've removed it earlier already. 🤔

Comment (greg0ire) [reply lib/Doctrine/ORM/Configuration.php]: Did a separate PR for that: https://github.com/doctrine/orm/pull/10321

File: lib/Doctrine/ORM/Configuration.php [hunk 3]
```diff
@@ -182,14 +183,21 @@ public function newDefaultAnnotationDriver($paths = [], $useSimpleAnnotationRead
         }
 
         if ($useSimpleAnnotationReader) {
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Configuration.php:185]: To be deprecated later on I suppose.

**Review by greg0ire — APPROVED** [review]

File: lib/Doctrine/ORM/Configuration.php (hunk 3 above)
Comment (derrabus) [reply lib/Doctrine/ORM/Configuration.php:185]: The whole method is deprecated already and its replacement does not have that parameter anymore.

**Pull request merged**